src/test/overlay/short_read_test.cpp

```diff
@@ -401,15 +405,17 @@ class short_read_test : public beast::unit_test::suite
             strand_type strand_;
             timer_type timer_;
             boost::asio::streambuf buf_;
+            endpoint_type const& ep_;
 
-            Connection (Client& client)
+            Connection (Client& client, endpoint_type const& ep)
```

Comment:
This interface takes a `const&` parameter and then stores that in a `const&` member variable. That makes it easy to store a reference to a temporary (`const&` will do lifetime extension). Consider changing the interface to take a raw pointer instead.